### PR TITLE
Move InstrumentationLibrary key generation logic out of TracerProvider/SpanConverter [#619]

### PR DESCRIPTION
**Pull request by ditsuke**
PR for Issue #619.

Factors key generation logic for InstrumentationLibrary instances out to a helper method in a utils class, replacing generation in `TracerProvider` and `SpanConverter` with calls to the new method.

### CODE REVIEW

**Comment by ditsuke** [comment]
Could use some feedback on the new `Helpers` class, can we do a better name?

**Comment by brettmc** [comment]
@ditsuke welcome aboard! re: class naming, I think `SDK\Common\Instrumentation` is a better location. As to class name, `KeyGenerator`? It will also need unit tests.

**Comment by ditsuke** [comment]
> @ditsuke welcome aboard! re: class naming, I think `SDK\Common\Instrumentation` is a better location. As to class name, `KeyGenerator`? It will also need unit tests.

@brettmc thanks! I've moved and renamed the generator and added unit tests, PTAL

File: tests/Unit/SDK/Common/Instrumentation/KeyGeneratorTest.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    public function test_null_version_schema(): void
+    {
+        $name = 'foo';
+
+        $key = KeyGenerator::generateInstanceKey($name, null, null);
+
+        $this->assertSame(sprintf('%s@unknown ', $name), $key);
+    }
```
Comment (tidal) [inline tests/Unit/SDK/Common/Instrumentation/KeyGeneratorTest.php:33]: Could you please also add a test for providing a `name`, `version` and no `schemaURL`?
This is only to assert, that key are different for all of these three variants.

**Comment by tidal** [comment]
@ditsuke 
Thank you. LGTM
I have added one little comment for a test you could add, just to make sure.
However other than his, I think we can merge the PR, if you are ok with it.

**Review by tidal — APPROVED** [review]

**Comment by ditsuke** [comment]
> @ditsuke Thank you. LGTM I have added one little comment for a test you could add, just to make sure. However other than t, I think we can merge the PR, if you are ok with it.

@tidal thanks for reviewing. I've added the third test with b84e3bb

**Comment by ditsuke** [comment]
Thanks!

**Pull request merged**